huff_assert/tests/runner.rs

```diff
@@ -0,0 +1,189 @@
+        &contract,
+        contract.macros.get(0).unwrap(),
+        String::default(),
+        U256::zero(),
+        vec![],
+    );
+
+    assert_eq!(res.name, "TEST");
+    assert!(res.errors.is_empty());
+    assert_eq!(res.reason, Return::Stop);
+}
+
+#[test]
+fn test_wrong_amount() {
+    let source = r#"
+    #define macro TEST() = takes (0) returns (0) {
+        0x10        $ [val]
+        pop         $ [val]
+    }
+    "#;
+
+    let contract = get_contract(source);
+
+    assert_eq!(contract.macros.len(), 1);
+
+    let res = inspect(
+        &contract,
+        contract.macros.get(0).unwrap(),
+        String::default(),
+        U256::zero(),
+        vec![],
+    );
+
+    assert_eq!(res.name, "TEST");
+    assert_eq!(res.errors.len(), 1);
+
+    let err = res.errors.get(0).unwrap();
+    assert_eq!(err.kind, ErrorKind::Amount);
+    assert_eq!(err.expected, "`[val]`");
+    assert_eq!(err.got, "`[]`");
+
+    assert_eq!(res.reason, Return::Stop); // didn't reverted
+}
+
+#[test]
+fn test_stack_underflow() {
+    let source = r#"
+    #define macro TEST() = takes (1) returns (0) {
+        0x10        $ [val]
+        pop pop     $ []
+    }
+    "#;
+
+    let contract = get_contract(source);
+
+    assert_eq!(contract.macros.len(), 1);
+
+    let res = inspect(
+        &contract,
+        contract.macros.get(0).unwrap(),
+        String::default(),
+        U256::zero(),
+        vec![],
+    );
+
+    assert_eq!(res.name, "TEST");
+    assert!(!res.errors.is_empty());
+    assert_eq!(res.reason, Return::StackUnderflow);
+}
+
+#[test]
+fn test_wrong_takes() {
+    let source = r#"
+    #define macro TEST() = takes (2) returns (0) {
+        0x10        $ [val]
+        pop         $ []
+    }
+    "#;
+
+    let contract = get_contract(source);
+
+    assert_eq!(contract.macros.len(), 1);
+
+    let res = inspect(
+        &contract,
+        contract.macros.get(0).unwrap(),
+        String::default(),
+        U256::zero(),
+        vec![],
+    );
+
+    assert_eq!(res.name, "TEST");
+    assert_eq!(res.errors.len(), 1);
+
+    let err = res.errors.get(0).unwrap();
+    assert_eq!(err.kind, ErrorKind::Takes);
+    assert_eq!(err.expected, "`takes(2)`");
+    assert_eq!(err.got, "`[]`");
+
+    assert_eq!(res.reason, Return::Stop); // didn't reverted
+}
+
+#[test]
+fn test_wrong_returns() {
+    let source = r#"
+    #define macro TEST() = takes (0) returns (0) {
+        0x10        $ [val]
+    }
+    "#;
+
+    let contract = get_contract(source);
+
+    assert_eq!(contract.macros.len(), 1);
+
+    let res = inspect(
+        &contract,
+        contract.macros.get(0).unwrap(),
+        String::default(),
+        U256::zero(),
+        vec![],
+    );
+
+    assert_eq!(res.name, "TEST");
+    assert_eq!(res.errors.len(), 1);
+
+    let err = res.errors.get(0).unwrap();
+    assert_eq!(err.kind, ErrorKind::Returns);
+    assert_eq!(err.expected, "`returns(0)`");
+    assert_eq!(err.got, "`[16]`");
```

Comment:
Should not use a string slice here, use the token representation or a string only surrounding `16`